app/components/submission/ReviewerSuggestions/FormSections.js

```diff
@@ -5,6 +5,14 @@ import { Checkbox } from '@pubsweet/ui'
 import ValidatedField from '../../ui/atoms/ValidatedField'
 import CalloutBox from '../../ui/atoms/CalloutBox'
 import Textarea from '../../ui/atoms/Textarea'
+import { FormH3 } from '../../ui/atoms/FormHeadings'
+
+const createCloseHandler = (index, values, fieldName, setFieldValue) => () =>
```

Comment:
We wouldn't have to dynamically generate a closeHandler, if we were to favour composition over configuration.

Having to "configure" `CalloutBox` to have an X icon/not have an X icon is what leads to this kind of extra abstraction to create closing functionality. IMO this is a little unnecessarily convoluted & makes this file slowly more and more difficult to reason about.

Reply:
It comes down to whether you think the panel component should know about the context in which it's being rendered (as part of an array in a form): if not, then you need something like this

Reply:
As a side note Tam poointed out we could simplify this in some respects by using Formik's `FieldArray`. I felt it was actually simpler to keep it this way though, as introducing `FormArray` creates complexities in other respects.